tools/generator/SourceWriters/MethodCallback.cs

```diff
@@ -79,6 +86,24 @@ protected override void WriteBody (CodeWriter writer)
 
 			if (!method.IsVoid && method.Parameters.HasCleanup)
 				writer.WriteLine ("return __ret;");
+
+			writer.Unindent ();
+
+			writer.WriteLine ("} catch (global::System.Exception __e) {");
+			writer.Indent ();
+			writer.WriteLine ("JniEnvironment.Runtime.OnUserUnhandledException (ref __envp, __e);");
```

Comment:
I *suspect* that this should probably use `global::Java.Interop.JniEnvironment.Runtime.OnUserUnhandledException (ref __envp, __e)`.